### PR TITLE
NPS Salesforce Integration

### PR DESCRIPTION
**Pull request by MatthewHightech**
The main goal of this PR was to integrate Salesforce Data such as FamilyMRR and the Customer Success Manager into the NPS reports. 

The following features have been verified that they work in the SMC preview environment through the Vercel staging link, and have been approved by stakeholder Adam Witt. 

New features achieved through this PR are:
- SalesforceDAO added to NPS
- MRR and Customer Success Manager displayed in NPS post
- Structure of the API requests have been adjusted with the mux routing library to support testing
- CORS headers were added through the mux routing library
- Channel ID's are stored in ENV variables, and are unique to the environment the app is deployed in

A Preview of what the NPS posts look like can be seen below. 
Screen Shot 2021-04-20 at 10 55 29 AM
Screen Shot 2021-04-20 at 10 49 57 AM

### CODE REVIEW

**Comment by MatthewHightech** [comment]
I didn't want to overwhelm this PR with too much content, but I guess I have to fix the test before it's approved...

File: api/nps/index.go [hunk 1]
```diff
@@ -90,26 +123,34 @@ func Handler(w http.ResponseWriter, r *http.Request) {
...
 		sendInternalServerError(w, err)
 		return
 	}
 }
 
-func createSlackAttachment(urlMap map[string][]string) (slack.Attachment, error) {
+func createSlackAttachment(urlMap map[string][]string, salesforceData []*salesforce.AccountInfo) (slack.Attachment, error) {
+	mrr, rep := "Unknown", "Unknown"
+	if len(salesforceData) > 0 {
+		mrr = "$" + formatInt(int(salesforceData[0].FamilyMRR))
```
Comment (codeallthethingz) [inline api/nps/index.go:151]: Might be cleaner to use the text package for this.

https://stackoverflow.com/a/46811454

File: api/nps/index.go [hunk 2]
```diff
@@ -90,26 +123,34 @@ func Handler(w http.ResponseWriter, r *http.Request) {
...
 
-	attachments, err := createSlackAttachment(urlMap)
+	attachments, err := createSlackAttachment(urlMap, responseData)
 	if err != nil {
 		sendInternalServerError(w, err)
 		return
 	}
 
-	err = slackDAO.sendSlackMessage(env.SlackOauthToken, attachments, os.Getenv("CHANNEL_ID"))
+	err = slackApi.sendSlackMessage(env.SlackOauthToken, attachments, os.Getenv("CHANNEL_ID"))
```
Comment (codeallthethingz) [inline api/nps/index.go:141]: what happens here if channel id is not set?

File: api/nps/index.go [hunk 3]
```diff
@@ -169,8 +225,21 @@ func createSlackAttachment(urlMap map[string][]string) (slack.Attachment, error)
...
+    startOffset := 3
+    if number < 0 {
+        startOffset++
+    }
+    for outputIndex := len(output); outputIndex > startOffset; {
+        outputIndex -= 3
+        output = output[:outputIndex] + "," + output[outputIndex:]
+    }
+    return output
+}
```
Comment (codeallthethingz) [inline api/nps/index.go:239]: listen to the tiny voice inside you that says "probably there's a library for this" :)

**Review by codeallthethingz — CHANGES_REQUESTED** [review]
looks good.  Couple changes!

File: api/nps/index.go (hunk 2 above)
Comment (MatthewHightech) [reply api/nps/index.go:141]: It throws an internal server error and logs "Channel_not_found"

**Pull request merged**